static/js/message_list_view.js

```diff
@@ -1445,4 +1445,76 @@ export class MessageListView {
             message_container.status_message = false;
         }
     }
+
+    update_sticky_recipient_headers() {
+        const rows_length = this._rows.size;
+        if (!rows_length) {
+            /* No headers are present */
+            return;
+        }
+        /* Intentionally remove sticky headers class here to make calculations simpler. */
+        $(".sticky_header").removeClass("sticky_header");
+        /* visible_top is navbar top position + height for us. */
+        const visible_top = $message_viewport.message_viewport_info().visible_top;
+        /* We need date to be properly visible on the header, so partially visible headers
+        who are about to be scrolled out of view are not acceptable. */
+        const $table = rows.get_table(this.table_name);
+        const $headers = $table.find(".message_header");
+        const header_height = $headers[0].getBoundingClientRect().height;
+
+        /* There are 3 types elements we expect to be present at this pixel:
+            * message header
+            * Date separator
+            * Z logo at the top of message feed
+
+            If we encounter either loading indicator or date separator, we set sticky header to the next recipient row. */
```

Comment:
Is there a way to write this algorithm in a way that doesn't depend on what other elements might be present there?

Reply:
No longer relevant.